docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1595,21 +1595,29 @@ The `ActionProfile` message includes the following fields:
   instance of a PSA Action Selector extern.
 
 * `size`, an `int64` representing the maximum number of member entries that the
-  Action Profile can hold, or, in the case of an Action Selector, the maximum
-  sum of all member weights across all selector groups.
-
-* `max_num_members`, an `int64` representing the maximum number of member
-  entries that an Action Profile or Action Selector can hold. PSA programs can
-  use the `@max_num_members` annotation to provide this value.  For Action
-  Profiles, this must be equal to the size. If the annotation is omitted, the
-  P4Info field will default to the same value as `size`.
+  Action Profile can hold. For Action Selectors, its semantics is specified by
+  the `selector_size_semantics` value as described below.
 
 * `max_group_size`, an `int32` which is 0 for an Action Profile, or, for an
-  Action Selector, represents the maximum sum of all member weights within any
-  given selector group. The `max_group_size` must be no larger than `size`. PSA
-  programs can use the `@max_group_size` annotation to provide this value for
-  Action Selectors. If the annotation is omitted, the P4Info field will default
-  to 0.
+  Action Selector, its semantics is specified by the
+  `selector_size_semantics` value as described below.
+  The `max_group_size` must be no larger than `size`. PSA programs can use the
+  `@max_group_size` annotation to provide this value for Action Selectors.
+  If the annotation is omitted, the P4Info field will default to 0.
+
+* `selector_size_semantics`, an enum for Action Selectors that
+  specifies how `size` and `max_group_size` are calculated. It can be either:
+    * `SUM_OF_WEIGHTS`, indicating that `size` and `max_group_size` represent
+      the maximum sum of weights that can be present across all selector groups
+      and within a single selector group respectively.
+    * `SUM_OF_MEMBERS`, indicating that `size` and `max_group_size` represent
+      the maximum number of members that can be present across all selector
+      groups and within a single selector group respectively.
+
+  PSA programs can use the `@selector_size_semantics` annotation with one of
```

Comment:
Just to confirm, the syntax we have in mind here is
@selector_size_semantics(SUM_OF_WEIGHTS)
@selector_size_semantics(SUM_OF_MEMBERS)
?

Reply:
Correct.